plugins/woocommerce-admin/client/marketplace/components/product-list-content/no-results.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import { useQuery } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import NoResultsIcon from '../../assets/images/no-results.svg';
+import { fetchDiscoverPageData, ProductGroup } from '../../utils/functions';
+import ProductLoader from '../product-loader/product-loader';
+import ProductList from '../product-list/product-list';
+import './no-results.scss';
+
+export default function NoResults(): JSX.Element {
+	const [ productGroup, setProductGroup ] = useState< ProductGroup >();
+	const [ isLoadingProductGroup, setisLoadingProductGroup ] =
+		useState( false );
+	const [ noResultsTerm, setNoResultsTerm ] = useState< string >( '' );
+
+	const query = useQuery();
+
+	useEffect( () => {
+		if ( query.term ) {
+			setNoResultsTerm( query.term );
+		}
+
+		if ( query.category ) {
+			setNoResultsTerm( query.category );
+		}
+	}, [ query ] );
+
+	useEffect( () => {
+		setisLoadingProductGroup( true );
+
+		fetchDiscoverPageData()
+			.then( ( products: Array< ProductGroup > ) => {
+				const mostPopularGroup = products.find(
+					( group ) => group.id === 'most-popular'
+				);
+
+				if ( ! mostPopularGroup ) {
+					return;
+				}
+
+				mostPopularGroup.items = mostPopularGroup.items.slice( 0, 9 );
+
+				setProductGroup( mostPopularGroup );
+			} )
+			.catch( () => {
+				setProductGroup( undefined );
+			} )
+			.finally( () => {
+				setisLoadingProductGroup( false );
+			} );
+	}, [] );
+
+	function renderProductGroup() {
+		if ( isLoadingProductGroup ) {
+			return <ProductLoader />;
+		}
+
+		if ( ! productGroup ) {
+			return <></>;
```

Comment:
Or just `</>`? I dunno if the linter will like that.

Reply:
Don't think a self-closing `</>` is valid React (without an opening `<>`: I imagine it can't tell the difference between a closing empty tag and a self-closing empty tag). But you might be able to return `null`, or an empty string, or something?

Reply:
Ah nice, fair enough. No biggie.